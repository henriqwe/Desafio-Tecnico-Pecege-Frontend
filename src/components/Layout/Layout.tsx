import { ReactNode } from 'react'
import styled from 'styled-components'

interface props {
  children: ReactNode
}
export function MainLayout({ children }: props) {
  return <Layout>{children}</Layout>
}
const Layout = styled.main`
  display: flex;
  flex: 1;
  width: 100vw;
  height: 100vh;
  overflow: none;
  justify-items: center;
  position: relative;
  transition: all 0.2s ease-in-out;
  @media (min-width: 992px) {
    padding-left: 20px;
    padding-right: 20px;
  }
  @media (min-width: 1024px) {
    padding-left: 100px;
    padding-right: 100px;
  }
  @media (min-width: 1200px) {
    padding-left: 200px;
    padding-right: 200px;
  }
`
