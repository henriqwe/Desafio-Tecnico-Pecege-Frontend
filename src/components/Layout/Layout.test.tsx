import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { MainLayout } from './Layout'

describe('Layout', () => {
  it('should render the contact list correctly', async () => {
    const { container, findByTestId } = render(
      <MainLayout>
        <div data-testid={'div-teste'}>teste</div>
      </MainLayout>
    )

    const divTeste = findByTestId('div-teste')
    expect(container).toMatchSnapshot()
    expect(divTeste).toBeTruthy()
  })
})
