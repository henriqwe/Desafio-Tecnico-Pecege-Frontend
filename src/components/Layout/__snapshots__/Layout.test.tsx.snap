// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render the contact list correctly 1`] = `
<div>
  <main
    class="sc-bcPKhP eYcKWU"
  >
    <div
      data-testid="div-teste"
    >
      teste
    </div>
  </main>
</div>
`;
